### ISSUE TITLE
Assertion `t >= 0 && t < n_classes` in Chesapeake13

### ISSUE THREAD

**Issue by patriksabol** [microsoft/torchgeo#1439]
### Description

When training on Chesapeake13 with num_classes=14 and ignore_index=0, an assertion error ‘t >= 0 && t < n_classes’ occurs. This indicates that there are more class indices in the Chesapeake label imagery than the specified num_classes. However, when running `gdalinfo Baywide_13Class_20132014.tif
  Min=1.000 Max=12.000 
  Minimum=1.000, Maximum=12.000,
`

Could there be a problem with computing indices from Chesapeake13? I am using RandomBatchGeoSampler, so this error could randomly appear when there is a patch with a false class ID.”

### Steps to reproduce

Training on Chesaspeake13

### Version

0.4.1

**Comment by adamjstewart** [comment]
I also noticed this here: https://github.com/microsoft/torchgeo/pull/1368#issuecomment-1565705950

@calebrob6 can you look into this?

**Comment by calebrob6** [comment]
Sure

**Comment by adamjstewart** [comment]
Any luck @calebrob6?

**Comment by adamjstewart** [comment]
Note that `Chesapeake13` no longer exists after #2214, but the question still applies for the remaining Chesapeake state-specific products.

**Comment by calebrob6** [comment]
`t >= 0 && t < n_classes` means that CrossEntropy is seeing class indices in your mask that are less than 0 or greater than number of classes - 1.

To actually use the new Chesapeake state-specific products in training the user will need to re-map the masks to the range [0, num_classes - 1].
